src/libYARP_dev/include/yarp/dev/IFrameSource.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_DEV_IFRAMESOURCE_H
+#define YARP_DEV_IFRAMESOURCE_H
+
+#include <unordered_set>
+#include <map>
+#include <vector>
+#include <functional>
+#include <robotology/Result.h>
+#include <yarp/dev/api.h>
+#include <yarp/math/Quaternion.h>
+#include <yarp/math/FrameTransform.h>
+
+namespace yarp {
+    namespace dev {
+        class IFrameSource;
+    }
+}
+
+struct CustomFrameCompare
+{
+    bool operator()(const yarp::math::FrameTransform& lhs, const yarp::math::FrameTransform& rhs)
```

Comment:
I may be wrong, but I guess that you need to mark as `inline` all the methods defined in the header, otherwise you can get an One Definition Rule (ODR) violation, see https://en.cppreference.com/w/cpp/language/definition . 
This comment applies also to the rest of the class in the headers.